envbuilder.go

```diff
@@ -156,6 +157,36 @@ func Run(ctx context.Context, options Options) error {
 		}
 	}
 
+	if options.GitURL != "" {
+		gitURLParsed, err := ParseGitURL(options.GitURL)
+		if err != nil {
+			return fmt.Errorf("invalid git URL: %w", err)
+		}
+		// If we're cloning over SSH, we need a known_hosts file.
+		if gitURLParsed.Scheme == "ssh" {
+			var knownHostsContent []byte
+			if options.GitSSHKnownHostsBase64 != "" {
+				if kh, err := base64.StdEncoding.DecodeString(options.GitSSHKnownHostsBase64); err != nil {
+					return fmt.Errorf("invalid known_hosts content: %w", err)
+				} else {
+					knownHostsContent = kh
+				}
```

Comment:
Normally I'd agree and prefer to avoiding `else`, but since `kh` is assigned in the `if`-statement, it needs to be in the `else` or then the assignment moved out of `if`. Both are alright approaches IMO.

Reply:
I'm going to simplify this logic a bit.